src/helpers/clipboard/clipboard_cells_state.ts

```diff
@@ -369,58 +381,51 @@ export class ClipboardCellsState extends ClipboardCellsAbstractState {
     const { sheetId, col, row } = target;
     const targetCell = this.getters.getEvaluatedCell(target);
 
-    if (clipboardOption?.pasteOption !== "onlyValue") {
-      const targetBorders = this.getters.getCellBorder(target);
-      const originBorders = origin.border;
-      const border = {
-        top: targetBorders?.top || originBorders?.top,
-        bottom: targetBorders?.bottom || originBorders?.bottom,
-        left: targetBorders?.left || originBorders?.left,
-        right: targetBorders?.right || originBorders?.right,
-      };
-      this.dispatch("SET_BORDER", { sheetId, col, row, border });
-    }
-    if (origin.cell) {
-      if (clipboardOption?.pasteOption === "onlyFormat") {
-        this.dispatch("UPDATE_CELL", {
-          ...target,
-          style: origin.cell.style,
-          format: origin.evaluatedCell.format,
-        });
-        return;
-      }
+    if (clipboardOption?.pasteOption === "onlyValue") {
+      const locale = this.getters.getLocale();
+      const content = formatValue(origin.evaluatedCell.value ?? "", { locale });
+      this.dispatch("UPDATE_CELL", { ...target, content });
+      return;
+    }
 
-      if (clipboardOption?.pasteOption === "onlyValue") {
-        const locale = this.getters.getLocale();
-        const content = formatValue(origin.evaluatedCell.value, { locale });
-        this.dispatch("UPDATE_CELL", { ...target, content });
-        return;
-      }
-      let content = origin.cell.content;
+    const targetBorders = this.getters.getCellBorder(target);
+    const originBorders = origin.border;
+    const border = {
+      top: targetBorders?.top || originBorders?.top,
+      bottom: targetBorders?.bottom || originBorders?.bottom,
+      left: targetBorders?.left || originBorders?.left,
+      right: targetBorders?.right || originBorders?.right,
+    };
+    this.dispatch("SET_BORDER", { sheetId, col, row, border });
 
-      if (origin.cell.isFormula && operation === "COPY") {
-        content = this.getters.getTranslatedCellFormula(
-          sheetId,
-          col - origin.position.col,
-          row - origin.position.row,
-          origin.cell.compiledFormula,
-          origin.cell.dependencies
-        );
-      }
+    if (clipboardOption?.pasteOption === "onlyFormat") {
```

Comment:
in this case, we will dispatch an UPDATE_CELL even if origin.cell is empty

Reply:
same comment concerning onlyValue

Reply:
up

Reply:
While I agree'd with Alexis at first time, it looks like displaying UPDATE_CELL with empty origin is a desired behavior as we have a test to ensure that copying format from an empty cell clear the format of the paste target. And same things happen when pasting the value from an empty cell to clear the content of a target cell